### ISSUE TITLE
[MPI_erweiterung_priority_052]: OsteoBioR. Save and load files fully describing models plus a text describing model 

### ISSUE THREAD

**Issue by jroachell15** [Pandora-IsoMemo/osteo-bior#1]
Original Ticket here: https://github.com/INWT/mpi-temporal-iso/issues/19

- [x] User will be able to save and load files fully describing models plus a text describing the model (text entered by user before saving) for apps ReSources /OsteoBioR /Bpred/ IsoMemo models. 
- [x] The saved files should be loadable directly into R so that a user may execute the models within the R environment 
- [x] (this should include the description of the different functions accessible via help). 
- [x] The user will also be able to load files available from a directory online (for the Desktop version of the apps this will require an internet connection). The files for each app/model will be stored in a separate online directory. These individual directories will be contained within a single directory. In the app interface files stored in an online directory are displayed in a pending list showing their names (without extension). The user selects one of these and presses load to load it.

---
- current online directory (inside this repository): https://github.com/Pandora-IsoMemo/osteo-bior/tree/feat/1_loadRemoteModel/inst/app/predefinedModels

**Comment by arunge** [comment]
[Feature](https://github.com/Pandora-IsoMemo/osteo-bior/pull/6#issue-1158493984) is available in the beta version.

**Comment by wahani** [comment]
Currently I see the following in the Logs:

```
Loading required package: shiny

Listening on http://0.0.0.0:3838

Attaching package: ‘colourpicker’

The following object is masked from ‘package:shiny’:

    runExample

Loading required package: Rcpp

Attaching package: ‘shinyMatrix’

The following object is masked from ‘package:OsteoBioR’:

    startApplication


Attaching package: ‘dplyr’

The following objects are masked from ‘package:stats’:

    filter, lag

The following objects are masked from ‘package:base’:

    intersect, setdiff, setequal, union

Loading required package: StanHeaders
rstan (Version 2.21.2, GitRev: 2e1f913d3ca3)
For execution on a local, multicore CPU with excess RAM we recommend calling
options(mc.cores = parallel::detectCores()).
To avoid recompilation of unchanged Stan programs, we recommend calling
rstan_options(auto_write = TRUE)
Warning in matrixInput(inputId = "dataMatrix", class = "numeric", value = matrix(ncol = 6,  :
  Copy paste functionality has been removed from the current version. Please use version 0.4.0 if needed.
Warning in matrixInput(inputId = "isotope", class = "numeric", value = matrix(ncol = 3,  :
  Copy paste functionality has been removed from the current version. Please use version 0.4.0 if needed.
Warning in matrixInput(inputId = "stayTimeMatrix", inputClass = "matrix-input-rownames",  :
  Copy paste functionality has been removed from the current version. Please use version 0.4.0 if needed.
Warning in matrixInput(inputId = "historicData", inputClass = "matrix-input-rownames",  :
  Copy paste functionality has been removed from the current version. Please use version 0.4.0 if needed.
Warning: Error in library: there is no package called ‘xlsx’
  53: <Anonymous>
Error in library(xlsx) : there is no package called ‘xlsx’
```

Sounds like there is a dependency missing. You can check this when you run the app locally:

```
docker pull ghcr.io/pandora-isomemo/osteo-bior:beta && docker run --network host --rm ghcr.io/pandora-isomemo/osteo-bior:beta
```

**Comment by arunge** [comment]
Thanks @wahani , now it runs again.

@isomemo could you test the feature ([on the beta](https://isomemoapp.com/app/osteo-bio-r-beta)) for loading a presaved model described here: https://github.com/Pandora-IsoMemo/osteo-bior/pull/6#issue-1158493984?
If all works as expected we can discuss how to proceed here (merge the current implementation to the main or seperate the folder into its own repository).

**Comment by arunge** [comment]
As just discussed with @isomemo :
- the upload option with the directory to store objects under https://github.com/Pandora-IsoMemo/osteo-bior/tree/beta/inst/app/predefinedModels is "better than expected"
- however, currently model inputs and data are not updated correctly when uploading a model
![image](https://user-images.githubusercontent.com/16759098/165294797-48b62482-2d52-4f59-9bde-b13d9e054659.png)
- [ ] include the option only to store data with model inputs, **without** the model output object itself -> requires a rerun of the model, but saves space on the online directory. Use the same folder for this.

**Comment by isomemo** [comment]
The above is correct but just as a reminder, the saving of modelling options must include also the data. Plus the user may only save these options/data or save these plus the modelling results.

**Comment by arunge** [comment]
The option
> Plus the user may only save these options/data or save these plus the modelling results.

is now available on the beta app https://isomemoapp.com/app/osteo-bio-r-beta :

![image](https://user-images.githubusercontent.com/16759098/170283721-4143261e-a203-4d5b-b890-1e0ee961fd35.png)

All is done here.

@isomemo Should we update the main app with the fixed up/download and the new option for saving only model inputs?

**Comment by isomemo** [comment]
Yes, please!

**Comment by arunge** [comment]
Merged version 22.5.2 into main.
